### PR TITLE
Add Dotchi-Loop

### PR DESCRIPTION
**Pull request by x-sheep**
https://www.nikoli.co.jp/en/puzzles/dotchi_loop.html

This puzzle type was promoted from Omopa to a full puzzle by Nikoli, so I added an implementation.

Rules:
* Draw a loop that goes through all white circles, and none of the black circles.
* Within a region, all white circles must either contain curves or straight lines.

Note that there is no restriction on how often the loop enters a region.

Example puzzle: https://pzprjs-git-fork-x-sheep-dotchi.robx.vercel.app/p.html?dotchi/5/6/0g1247o30069al066ab0

### CODE REVIEW

File: src/variety/country.js [hunk 1]
```diff
@@ -165,15 +176,20 @@
 			}
 		}
 	},
+	"Cell@dotchi#1": {
```
Comment (tausbn) [inline src/variety/country.js:179]: It’s been a while since I was familiar with the object system of `pzprjs`. Can you remind me what `#1` signifies?

**Review by tausbn — APPROVED** [review]
This looks good to me (though my knowledge of `pzprjs` is a bit rusty).

One thing I find a bit curious is the fact that there’s no check that the shaded circles are not intersected by the loop. I realise that this is likely because it’s simply not possible to draw the loop across the shaded circles, but it _is_ possible to do so during editing (by placing the loop first and the shaded circle second).

The only argument I can think of in favour of allowing loop segments that intersect shaded circles is that it would make it slightly more flexible for variants, but I guess one might as well use `moonsun` for that.

File: src/variety/country.js (hunk 1 above)
Comment (x-sheep) [reply src/variety/country.js:179]: If I leave out the `#1`, this declaration would conflict with `Cell@moonsun,dotchi` above. pzprjs strips the puzzle names after the `@`.

Comment (tausbn) [reply src/variety/country.js:179]: Ah, I understand now. Thanks for the explanation.

**Pull request closed without merging**